transaction/_transaction.py

```diff
@@ -760,3 +753,13 @@ def __init__(self, datamanager):
 
     def rollback(self):
         raise TypeError("Savepoints unsupported", self.datamanager)
+
+def text_or_warn(s):
+    if isinstance(s, text_type):
+        return s
+
+    warnings.warn("Expected text", DeprecationWarning, stacklevel=3)
```

Comment:
Unfortunately, this will cause memory leaks in "warning-once" mode, because the text is part of the key to remember whether a warning was issued or not. This applies to both the builtin (_warnings) and python (warnings) versions.